eth/watchers/senderwatcher.go

```diff
@@ -0,0 +1,248 @@
+package watchers
+
+import (
+	"fmt"
+	"log"
+	"math/big"
+	"sync"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/golang/glog"
+	"github.com/livepeer/go-livepeer/eth"
+	"github.com/livepeer/go-livepeer/eth/blockwatch"
+	"github.com/livepeer/go-livepeer/eth/contracts"
+	"github.com/livepeer/go-livepeer/pm"
+)
+
+// SenderWatcher maintains a concurrency-safe map with SenderInfo
+// and implements the SenderManager interface
+type SenderWatcher struct {
+	senders map[ethcommon.Address]*pm.SenderInfo
+	mu      sync.RWMutex
+	quit    chan struct{}
+	watcher BlockWatcher
+	lpEth   eth.LivepeerEthClient
+	dec     *EventDecoder
+}
+
+// NewSenderWatcher initiates a new SenderWatcher
+func NewSenderWatcher(addr ethcommon.Address, watcher BlockWatcher, lpEth eth.LivepeerEthClient) (*SenderWatcher, error) {
+	dec, err := NewEventDecoder(addr, contracts.TicketBrokerABI)
+	if err != nil {
+		return nil, err
+	}
+	return &SenderWatcher{
+		quit:    make(chan struct{}),
+		watcher: watcher,
+		lpEth:   lpEth,
+		senders: make(map[ethcommon.Address]*pm.SenderInfo),
+		dec:     dec,
+	}, nil
+}
+
+// GetSenderInfo returns the senderInfo for a sender
+// if values for a sender are not cached an RPC call to a remote ethereum node will be made to initialize the cache
+func (sw *SenderWatcher) GetSenderInfo(addr ethcommon.Address) (*pm.SenderInfo, error) {
```

Comment:
Instead of creating an additional map, could we instead check if the key exists in `SenderWatcher.senders` when handling logs? 

My thought process being: 

For use of this watcher on the broadcaster side, there will always only be one entry (B's address) present in the map after the first `SenderWatcher.GetSenderInfo()` call. 

For use of this watcher on the orchestrator side we will initialize a map entry for a new sender upon our first call of `SenderWatcher.GetSenderInfo()` at which point we'll start handling events pertaining to that sender. When that map entry gets removed by the SenderMonitor cleanup loop (TODO!) we will no longer process events for that sender.